instrumentation/redisson/redisson-common/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/redisson/AbstractRedissonAsyncClientTest.java

```diff
@@ -0,0 +1,253 @@
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("SET")
+                        .hasKind(CLIENT)
+                        .hasAttribute(SemanticAttributes.NET_SOCK_PEER_ADDR, "127.0.0.1")
+                        .hasAttribute(SemanticAttributes.NET_SOCK_PEER_NAME, "localhost")
+                        .hasAttribute(SemanticAttributes.NET_SOCK_PEER_PORT, (long) port)
+                        .hasAttribute(SemanticAttributes.DB_SYSTEM, "redis")
+                        .hasAttribute(SemanticAttributes.DB_STATEMENT, "SET foo ?")
+                        .hasAttribute(SemanticAttributes.DB_OPERATION, "SET")));
+  }
+
+  @Test
+  void futureWhenComplete() throws ExecutionException, InterruptedException, TimeoutException {
+    RSet<String> set = redisson.getSet("set1");
+    CompletionStage<Boolean> result =
+        testing.runWithSpan(
+            "parent",
+            () -> {
+              RFuture<Boolean> future = set.addAsync("s1");
+              return future.whenComplete(
+                  (res, throwable) -> {
+                    assertThat(Span.current().getSpanContext().isValid()).isTrue();
+                    testing.runWithSpan("callback", () -> {});
+                  });
+            });
+    result.toCompletableFuture().get(30, TimeUnit.SECONDS);
+
+    testing.waitAndAssertSortedTraces(
+        orderByRootSpanName("parent", "SADD", "callback"),
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasName("parent").hasKind(INTERNAL).hasNoParent(),
+                span ->
+                    span.hasName("SADD")
+                        .hasKind(CLIENT)
+                        .hasAttribute(SemanticAttributes.NET_SOCK_PEER_ADDR, "127.0.0.1")
+                        .hasAttribute(SemanticAttributes.NET_SOCK_PEER_NAME, "localhost")
+                        .hasAttribute(SemanticAttributes.NET_SOCK_PEER_PORT, (long) port)
+                        .hasAttribute(SemanticAttributes.DB_SYSTEM, "redis")
+                        .hasAttribute(SemanticAttributes.DB_STATEMENT, "SADD set1 ?")
+                        .hasAttribute(SemanticAttributes.DB_OPERATION, "SADD")
+                        .hasParent(trace.getSpan(0)),
+                span -> span.hasName("callback").hasKind(INTERNAL).hasParent(trace.getSpan(0))));
+  }
+
+  // regression test for
+  // https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/6033
+  @Test
+  void scheduleCallable()
+      throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+    RScheduledExecutorService executorService = redisson.getExecutorService("EXECUTOR");
+    Object o = invokeSchedule(executorService);
+    Object getTaskId = o.getClass().getMethod("getTaskId").invoke(o);
+    assertThat(getTaskId).isNotNull();
+  }
+
+  private static Object invokeSchedule(RScheduledExecutorService executorService)
```

Comment:
Just wondering: does this method change in different versions of redisson? Why use reflection here?

Reply:
I have added a comment to explain it.  But I'm not sure the grammar is correct.

Reply:
By the way, should I rebase the branch to combine commits into one?  I think the merge work could do it as well